kafka/producer.go

```diff
@@ -287,10 +287,9 @@ func channelProducer(p *Producer) {
 
 // channelBatchProducer serves the ProduceChannel channel and attempts to
 // improve cgo performance by using the produceBatch() interface.
-func channelBatchProducer(p *Producer) {
+func channelBatchProducer(p *Producer, batchSize int) {
```

Comment:
I'm not sure it is warranted, this is only an internal optimization in the Go bindings, this batching does not correlate to the application nor librdkafka internals or Kafka protocol batches.